src/core/server/http/router/router.ts

```diff
@@ -270,7 +270,19 @@ export class Router<Context extends RequestHandlerContext = RequestHandlerContex
 
     try {
       const kibanaResponse = await handler(kibanaRequest, kibanaResponseFactory);
-      return hapiResponseAdapter.handle(kibanaResponse);
+      const handledResponse = hapiResponseAdapter.handle(kibanaResponse);
+      // Ensure similar "Internal Server Error" behaviour when the handler throws
+      // and when it returns a response with 500 error.
+      // NB: If 500 error, the previous steps ensures it's a Boom error,
+      // unless it's a streamed response and we should not intercept it
+      if (
+        !kibanaResponse.customResponse && // Except for custom responses (built via res.custom)
```

Comment:
I tried to explain it in the description:

> The reason for allowing `500` errors to be unchanged in the `res.custom({...})`-built responses is because `reporting` uses it to forward the actual errors reported by the jobs.

I can add `customError` to behave the same as `custom`, or just document the special behaviour of `custom`. What do you think?

Reply:
Interestingly, it answers `500` even if it's an expected situation. Even a comment here hints:
https://github.com/elastic/kibana/blob/eba61f9241aa1e39ed0cf25ac218cd1601e410b2/x-pack/plugins/reporting/server/routes/lib/get_document_payload.ts#L79-L80

ok, let's document this behavior as an escape hatch for the `custom` response factory. 
I searched through the Kibana repo there are lots of places using `custom` when it's completely unnecessary. 
https://github.com/elastic/kibana/blob/eba61f9241aa1e39ed0cf25ac218cd1601e410b2/x-pack/plugins/fleet/server/routes/epm/handlers.ts#L127
https://github.com/elastic/kibana/blob/eba61f9241aa1e39ed0cf25ac218cd1601e410b2/x-pack/plugins/maps/server/routes.js#L531

We need to switch them to the pre-defined `response factory` methods.